src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/AStar.java

```diff
@@ -0,0 +1,96 @@
+package edu.greenblitz.tobyDetermined.Nodesssss;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import java.util.LinkedList;
+
+public class AStar {
+    static LinkedList<NodeArm> open = new LinkedList<>();
+    static LinkedList<NodeArm> closed = new LinkedList<>();
+    static NodeBase nodeBase = NodeBase.getInstance();
+
+   public static double getDistanceFromCurrentNodeToEndOrStartNode(NodeArm point, NodeArm current){
+       double distance =  nodeBase.getDistanceBetweenToPoints(point,current);
+       if(distance < 0){
+           distance*=-1;
+       }
+       return distance;
+   }
+    public static double getDistanceToStartPlusEnd(NodeArm current, NodeArm start, NodeArm end){
+       double gCost = getDistanceFromCurrentNodeToEndOrStartNode(start,current);
+       double hCost = getDistanceFromCurrentNodeToEndOrStartNode(end,current);
+       return gCost+hCost;
+    }
+
+
+    public static NodeArm getLowestFcost(LinkedList<NodeArm> open, NodeArm start, NodeArm end) {
+       if (open.isEmpty()) {
+           return null; // Handle the case where the list is empty
+       }
+       int saveI = 0;
+       double fCost = getDistanceToStartPlusEnd(open.get(0), start, end);
+       for (int i = 1; i < open.size(); i++) {
+           double currentFCost = getDistanceToStartPlusEnd(open.get(i), start, end);
+           if (currentFCost < fCost) {
+               fCost = currentFCost;
+               saveI = i;
+           }
+       }
+       return open.get(saveI);
+   }
+
+
+   public static boolean isInList(NodeArm nodeArm, LinkedList<NodeArm> list){
+       for (NodeArm arm : list) {
+           if (nodeArm.getId() == arm.getId())
+               return true;
+       }
+       return false;
+    }
+
+    public static LinkedList<NodeArm> printPathPath(NodeArm nodeArm){
```

Comment:
Naming(?)

Reply:
To clarify (as I understood), the function name says print but the function doesn’t actually print, it returns the path.
Correct me if I’m wrong

Reply:
Functions names should be just clearer - using the word path twice isn't that good either ;)

Reply:
changed to 'printAndReturnPath'